test/ex_oauth2_provider/oauth2/authorization/strategy/code_test.exs

```diff
@@ -83,6 +83,12 @@ defmodule ExOauth2Provider.Authorization.CodeTest do
       %{resource_owner: resource_owner, application: application}
     end
 
+    test "with no scope", %{resource_owner: resource_owner, application: application} do
```

Comment:
This is tested with an application that doesn't have any scope, but relies on the server scope. Is this as intended?

Reply:
Yes hmm - I think that is fair.  I really appreciate your thoughts
I think in general that part of the trouble OAUTH2 creates for itself is not having strong and clear best options to follow. IE (PKCE + authorization code will cover 95% of use cases , but instead it has password grant (only secure for specific cases) , credential grant, implicit grant (depricated) )

Maybe the right question we need to be asking is : Is a token with no scope legitimate if the application specifies a scope?